### ISSUE TITLE
Scheduled layouts not working

### ISSUE THREAD

**Issue by jetski28** [xibosignage/xibo#246]
During our testing stage we were running a fairly vanilla setup of XIBO 1.6.3 both Server (Linux) & Client (Windows) and recently put it into production with 6 clients. All working well with a simple scheduled layout (3 Regions - Logo, Date & Notification Area)  to run 8 hours each day until today when our clients were only showing the default layout (Background image with a text clock).

No matter what I tried, no layout would appear on the clients including a simple layout with one region of testing text. If I set any of the layouts to be the default layout on the clients it would appear without an issue. All the layouts are valid and ready to play.

The date & time between the server & clients are all in time and we have the client collection interval set to 30 seconds - this rules out any timing issues.

Basically now I have ran out of ideas after reading several support posts on similar issues like I have that I decided to upgrade to XIBO 1.7.1 (Server) & 1.7.0 (Client) including a clean install which has made no difference what so ever.

The logs on the clients are showing me no errors in audit mode and just a continuious log of the default layout expiring and starting again each minute (60 second duration) with the download of the Schedule XML every 30 seconds. All clients are licensed and have a valid status within CMS.

ScheduleManagerThread|13/02/2015 10:07:47 PM|Audit|ScheduleManager - Run|Schedule Timer Ticked
RegisterAgentThread|13/02/2015 10:07:48 PM|Info|RegisterAgent - Run|Thread Woken and Lock Obtained
UI Thread|13/02/2015 10:07:53 PM|Audit|Media - SignalElapsedEvent|Media Complete
UI Thread|13/02/2015 10:07:53 PM|Audit|Region - DurationElapsedEvent|Media Elapsed: 
UI Thread|13/02/2015 10:07:53 PM|Audit|Region - SetNextMediaNode|Media Expired:(578,146,928,1336,text,) . Reached the end of the sequence. Starting from the beginning.
UI Thread|13/02/2015 10:07:53 PM|Audit|MainForm - ScheduleChangeEvent|Schedule Changing to C:\Users\Andrew\Documents\Xibo Library\5.xlf
UI Thread|13/02/2015 10:07:53 PM|Audit|Region - SetNextMediaNode|New media detected text
UI Thread|13/02/2015 10:07:53 PM|Audit|Region - CreateNextMediaNode|Creating new media: text, 11846d5d9f686fb75fc9dad0b19ca9de
UI Thread|13/02/2015 10:07:53 PM|Audit|Region - StartMedia|Starting media
ScheduleManagerThread|13/02/2015 10:07:57 PM|Audit|ScheduleManager - Run|Schedule Timer Ticked
ScheduleManagerThread|13/02/2015 10:08:07 PM|Audit|ScheduleManager - Run|Schedule Timer Ticked
ScheduleAgentThread|13/02/2015 10:08:12 PM|Info|ScheduleAgent - Run|Thread Woken and Lock Obtained
ScheduleAgentThread|13/02/2015 10:08:12 PM|Audit|ScheduleManager - GetScheduleXmlString|Getting the Schedule XML
RequiredFilesAgentThread|13/02/2015 10:08:13 PM|Audit|RequiredFiles - SetRequiredFiles|Building required file node for 9
RequiredFilesAgentThread|13/02/2015 10:08:13 PM|Audit|RequiredFiles - SetRequiredFiles|Building required file node for 10
RequiredFilesAgentThread|13/02/2015 10:08:13 PM|Audit|RequiredFiles - SetRequiredFiles|Building required file node for 11
RequiredFilesAgentThread|13/02/2015 10:08:13 PM|Audit|RequiredFiles - SetRequiredFiles|Building required file node for 12
RequiredFilesAgentThread|13/02/2015 10:08:13 PM|Audit|RequiredFiles - SetRequiredFiles|Building required file node for 13
RequiredFilesAgentThread|13/02/2015 10:08:13 PM|Audit|RequiredFiles - SetRequiredFiles|Building required file node for 14
RequiredFilesAgentThread|13/02/2015 10:08:13 PM|Audit|RequiredFiles - SetRequiredFiles|Building required file node for 15
RequiredFilesAgentThread|13/02/2015 10:08:13 PM|Audit|RequiredFiles - SetRequiredFiles|Building required file node for 16
RequiredFilesAgentThread|13/02/2015 10:08:13 PM|Audit|RequiredFiles - SetRequiredFiles|Building required file node for 17
RequiredFilesAgentThread|13/02/2015 10:08:13 PM|Audit|RequiredFiles - SetRequiredFiles|Building required file node for 19
RequiredFilesAgentThread|13/02/2015 10:08:13 PM|Audit|RequiredFiles - SetRequiredFiles|Building required file node for 5

On the server I see the following in the logs for the CMS but not sure whether this is related to my issue.

<errormsg>Undefined variable: type</errormsg>
<errornum>8</errornum>
<errortype>Notice</errortype>
<scriptname>/var/www/html/xibo/lib/service/xmdssoap4.class.php</scriptname>
<scriptlinenum>932</scriptlinenum>

Am I missing something or is something gone wrong?
Any ideas would be greatly appreciated.

**Comment by dasgarner** [comment]
A screenshot of the player information screen would help if possible? (press 'i' on the keyboard when at the player pc).

The info screen shows the schedule the player thinks it should be playing - if that only shows the default layout then we are at least pointed towards something in the CMS.


If that is the case - we will need to enable auditing for that display in the CMS Displays page (edit display, advanced tab) and then enable global auditing by changing the log type in CMS settings.

We can then look at the CMS log page, filtered by your Display for the function "Schedule". It should show us the schedule that is being resolved and the date/times that are being used to resolve it.

**Comment by jetski28** [comment]
Here is the screenshot as requested...
![Uploading xibo-client.png . . .]()

**Comment by jetski28** [comment]
Here is the log from the CMS for one of the displays...
![xibo-schedule-log](https://cloud.githubusercontent.com/assets/10990419/6198420/023659ca-b454-11e4-9f9f-28528880ef9f.png)

**Comment by jetski28** [comment]
I think I have just solved my problem - during our testing stage I had several clients under one single name and client ID as a master image was created to deploy out to the hardware for the other clients.

Just recently I noticed this and changed the client ID slightly of each client so they showed up as invididual items within the CMS. As the testing stage went on for some months I cannot recall making the "All Displays" group and (you might be able to correct me) I assumed it was there by default and had all displays listed automatically.

When I did a clean install of XIBO 1.7.1 I did notice that not one display group existed and had to create it manually. I was certain I added all clients to this group but apparently not. Once I assigned them to the group, all systems go.

Silly me all along - and I thought I was going crazy!
